test/utils/test_signer.ts

```diff
@@ -0,0 +1,160 @@
+import { SupportedProvider } from '@0x/order-utils';
+import {
+    NonceTrackerSubprovider,
+    PartialTxParams,
+    PrivateKeyWalletSubprovider,
+    RPCSubprovider,
+    Web3ProviderEngine,
+} from '@0x/subproviders';
+import { BigNumber, providerUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { utils as web3WrapperUtils } from '@0x/web3-wrapper/lib/src/utils';
+import EthereumTx = require('ethereumjs-tx');
+import { Connection, Repository } from 'typeorm';
+
+import { ETH_TRANSFER_GAS_LIMIT, EXPECTED_MINED_SEC } from '../../src/constants';
+import { TransactionEntity } from '../../src/entities';
+import { logger } from '../../src/logger';
+import { TransactionStates } from '../../src/types';
+import { TEST_RINKEBY_CHAIN_ID } from '../config';
+
+const DEFAULT_TX_VALUE = 1337;
+const HIGH_GAS_PRICE_FOR_UNSTICKING = 13000000000;
+
+export class TestSigner {
```

Comment:
Hmmm this is a pretty complex mock, does it need to be this complex? why can't we use the real signer?

Reply:
This `TestSigner` was used during the development to quickly iterate and keep the test simpler.

After we will have the full integration test for meta transactions we might just delete this mock altogether.